lib/recipe.rb

```diff
@@ -0,0 +1,24 @@
+class Recipe
+attr_reader :name,
+            :ingredients_required
+
+  def initialize(name)
+    @name = name
+    @ingredients_required = Hash.new
+  end
+
+  def add_ingredient(ingredient, amount)
+    (@ingredients_required[ingredient] += amount if @ingredients_required[ingredient]) ||
```

Comment:
Could have also just done a default value on your hash like you did with the other class.